From c6e5359374ee9b46777be790da9d64d7c408a5c1 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <bonzini@gnu.org>
Date: Tue, 5 Jul 2011 10:50:58 +0200
Subject: [PATCH 1/1] fix typo in 2fc74b9

2011-08-13  Paolo Bonzini <bonzini@gnu.org>

	* scripts/Remote.st: Fix typo in 2fc74b9.
---
 ChangeLog         |    4 ++++
 scripts/Remote.st |   10 +++++++---
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/scripts/Remote.st b/scripts/Remote.st
index ec65de4..cf383bd 100644
--- a/scripts/Remote.st
+++ b/scripts/Remote.st
@@ -182,11 +182,14 @@ available in the PATH of the remote machine.
     opt isNil ifTrue: [
 	host isNil ifFalse: [ self error: 'multiple hosts are invalid' ].
 	(arg includes: $@)
-	    ifFalse: [ host := TCP.SocketAddress byName: arg ]
+	    ifFalse: [ host := arg ]
 	    ifTrue: [
 		login isNil ifFalse: [ self error: 'multiple logins are invalid' ].
-		login := TCP.SocketAddress byName: (arg copyUpTo: $@).
-		host := arg copyAfter: $@ ] ].
+		login := arg copyUpTo: $@.
+		host := arg copyAfter: $@ ].
+	(TCP.SocketAddress byName: host)
+	    ifNil: [ self error: 'invalid host %1' %{host} ]
+            ifNotNil: [ :addr | host := addr ] ].
     ]
 
     ifError: [
@@ -197,6 +200,7 @@ available in the PATH of the remote machine.
         ('gst-remote: ', ex messageText, '
 ') displayOn: stderr.
 	stderr flush.
+        "ex pass."
 	helpString displayOn: stderr.
 	ObjectMemory quit: 1 ].
 
-- 
1.7.2.5

